### PR TITLE
[pt-br] JavaScript Essencials

### PR DESCRIPTION
**Pull request by helcioItiyama**
Hi guys! I translated a few more files @khaosdoctor, @alvarocamillont, @danilofuchs @orta

### CODE REVIEW

File: packages/playground-examples/copy/pt/JavaScript/JavaScript Essentials/Code Flow.ts [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+//Se você passar o mouse por cima dos três casos de uso de 'jon' abaixo,
+//você verá que a tipagem muda dependendo de onde a palavra está alocada:
+
+if (jon) {
+  jon;
+} else {
+  jon;
+}
+
+//O tipo '{name: string } | undefined' usa uma ferramenta do TypeScript
```
Comment (danilofuchs) [inline packages/playground-examples/copy/pt/JavaScript/JavaScript Essentials/Code Flow.ts]:
```suggestion
//O tipo '{ name: string } | undefined' usa uma ferramenta do TypeScript
```

File: packages/playground-examples/copy/pt/JavaScript/JavaScript Essentials/Functions.ts [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+//A seguir temos uma versão simples da função de adicionar,
+//a qual recebe dois parametros de qualquer tipagem (any): 
+//poderiam ser passadas strings, números ou objetos.
+
+const add1 = (x: any, y: any) => {
+  return x + y;
+};
+add1("Hello", 23);
+
+//Isso é legítimos em JavaScript (strings pode ser
```
Comment (danilofuchs) [inline packages/playground-examples/copy/pt/JavaScript/JavaScript Essentials/Functions.ts]:
```suggestion
//Isso é legítimo em JavaScript (strings podem ser 
```

File: packages/playground-examples/copy/pt/JavaScript/JavaScript Essentials/Functions.ts [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+//não cumpriu com o esperado.
+
+const add4 = (x: number, y: number): number => {
+  return x + y;
+};
+
+//Isso é uma versão mais explícita de add2 - Existem casos
+//em que você talvez queira usar uma sintaxe explícita do tipo 
+//de retorno para dar espaço para trabalhar antes de começar.
+//Um pouco como o desenvolvimento orientado a testes recomenda
```
Comment (danilofuchs) [inline packages/playground-examples/copy/pt/JavaScript/JavaScript Essentials/Functions.ts]:
```suggestion
//Um pouco como o desenvolvimento orientado a testes (TDD) recomenda
```

File: packages/playground-examples/copy/pt/JavaScript/JavaScript Essentials/Objects and Arrays.ts [hunk 4]
```diff
@@ -0,0 +1,110 @@
+//// { order: 1, compiler: { strict: false } }
+
+//Objetos em JavaScript são coleções de valores agrupadas
```
Comment (danilofuchs) [inline packages/playground-examples/copy/pt/JavaScript/JavaScript Essentials/Objects and Arrays.ts]:
```suggestion
//Objetos em JavaScript são coleções de valores agrupados
```

File: packages/playground-examples/copy/pt/JavaScript/JavaScript Essentials/Objects and Arrays.ts [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+//informações adicionais sem precisar procurar em outro local
+
+//TypeScript também oferece esses mesmos recursos para arrays.
+//Aqui temos um array com apenas nossos pedidos de compra acima.
+
+const allOrders = [purchaseOrder];
+
+//Se você passar o mouse por cima de allOrders, você poderá
+//afirmar que é um array, porque as informações passadas pelo
+// cursor terminam com um []. Você pode acessar o primeiro pedido
```
Comment (danilofuchs) [inline packages/playground-examples/copy/pt/JavaScript/JavaScript Essentials/Objects and Arrays.ts]:
```suggestion
//cursor terminam com um []. Você pode acessar o primeiro pedido
```

File: packages/playground-examples/copy/pt/JavaScript/JavaScript Essentials/Objects and Arrays.ts [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+//
+//  purchaseOrder - Obtem a variável purchaseOrder e informa ao 
+//                  TypeScript que este é o formato de todos
+//                  os objetos no array de pedidos.
+//
+//  readonly - Este objeto não permite mutação, uma vez criado,
+//             o conteúdo do array sempre permanecerá o mesmo.
+//
+//
+// Agora, se você tentar utilizar o método pop() do array readonlyOrders,
```
Comment (danilofuchs) [inline packages/playground-examples/copy/pt/JavaScript/JavaScript Essentials/Objects and Arrays.ts]:
```suggestion
//Agora, se você tentar utilizar o método pop() do array readonlyOrders,
```

**Review by danilofuchs — CHANGES_REQUESTED** [review]
Great job!
Only a few typos

**Comment by orta** [comment]
Great, thanks!

**Pull request merged**